service/mapper/status_test.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package mapper
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStatus_String(t *testing.T) {
+	assert.Equal(t, StatusEmpty.String(), "empty")
+	assert.Equal(t, StatusUpdating.String(), "updating")
+	assert.Equal(t, StatusMatched.String(), "matched")
+	assert.Equal(t, StatusCollected.String(), "collected")
+	assert.Equal(t, StatusIndexed.String(), "indexed")
+	assert.Equal(t, StatusForwarded.String(), "forwarded")
+
+	assert.Equal(t, Status(0).String(), "invalid")
```

Comment:
I don't think this should be tested, as it is meant for human consumption. You seem to have been super eager to just test every single line of code :D

Reply:
When it doesn't cost much, I like to do it yep! In most cases it's useless, but in some cases, if the code that we're testing changes unexpectedly, it breaks a test and might help to find an unwanted change or a mistake :p

Reply:
In this case, I would prefer to remove it. We should not have failing tests when changing the strings here imo.

Reply:
It's not about changing the strings, it's about changing the behavior. If we were to rewrite this func and forget to handle the case where an invalid status is given, this will let us know.

For me, unit tests are a lot about bringing attention to unexpected changes. Sometimes when changing code, we might not realize that we're changing a behavior that someone relies upon. That's why I like to test every little detail, so that when we change them, we need to change the tests so it brings our attention to it. _Is it really a good idea to remove/change this bit of code?_ / _Oh shit, I forgot that this was used in package X, let me put it back._

Reply:
Even _if_ that were to happen, it would _not_ affect the code. Nothing would break. There is no _behaviour_ here. The code only exists to provide humans with more readable output. I don't think that should be tested, because there is no logically _correct_ way to do so.

Reply:
Your argument about unit tests doesn't really apply here. Even if every single status would print as `undefined`, nothing would be broken, except for output intended for human consumption. It's the equivalent of testing the error string of every single error type in the code base. It makes no sense and is, imo, actually an anti-pattern.

Reply:
Your argument about unit tests doesn't really apply here. Even if every single status would print as undefined, nothing would be broken, except for output intended for human consumption. It's the equivalent of testing the error string of every single error type in the code base. It makes no sense and is, imo, actually an anti-pattern. Or are you saying it would be a good idea?

Reply:
We intend it to be for human consumption but IMO we shouldn't assume how consumers of our package might use it, in an open source project. Some random other project might be using this and for some reason they might rely on this 🤷 If this was closed-source I'd completely agree, but here I feel like it can be useful. If you guys still think it's not worth it, I'm fine with removing it though 👍

Reply:
I think I'd also vote in not covering it with tests in this case. We could always document the `String()` method of `Status` that it's debug/internal use only and shield ourselves from any improper use..

That's something I've seen around multiple libraries before, even though it's a bit ugly on the eyes.

Reply:
> ... Some random other project might be using this and for some reason they might rely on this.

If they do this, I would rather want to implement code that breaks their code _on purpose_, because there should absolutely never be any project that relies on this. Just like no project should ever use an error string to check for the type of an error, and if they do, they should be punished :P.

Reply:
Test removed then 👍